src/main/java/com/upsolver/datasources/jdbc/querybuilders/OracleQueryDialect.java

```diff
@@ -41,4 +59,31 @@ public boolean isAutoIncrementColumn(ResultSet columnsResultSet) throws SQLExcep
         // Example default value: "ADMIN"."ISEQ$$_20599".nextval
         return def != null && def.toUpperCase().endsWith(".NEXTVAL") && def.toUpperCase().contains("ISEQ$$");
     }
+
+
+    @Override
+    public SQLType getSqlType(int code) throws SQLException {
+        return Optional.ofNullable((SQLType)OracleType.toOracleType(code)).orElseGet(() -> JDBCType.valueOf(code));
+    }
+
+
+    @Override
+    public boolean isTimeType(SQLType sqlType) throws SQLException {
+        return oracleTimeTypes.contains(sqlType) || super.isTimeType(sqlType);
+    }
+
+    @Override
+    public SQLType getJdbcType(SQLType sqlType) {
+        if (sqlType instanceof OracleType) {
+            Integer oracleCode = sqlType.getVendorTypeNumber();
+            if (oracleCode != null) {
+                try{
+                    return JDBCType.valueOf(oracleCode);
+                } catch (IllegalArgumentException e) {
+                    return sqlType;
+                }
+            }
+        }
+        return sqlType;
```

Comment:
Why convert if the original SQLType works fine?

Reply:
I've added comment that explains the reason to the code.